src/utils.h

```diff
@@ -44,5 +46,34 @@ std::set<std::string> getMapKeys(const T& map) {
     });
     return ret;
 }
+
+using json = nlohmann::json;
+
+inline int64_t get_int64_or_throw(const json& el) {
+    if (!el.is_number()) {
+        throw SonataError(fmt::format("expected integer, got {}", el.dump()));
+    }
+    auto v = el.get<double>();
+    if (std::floor(v) != v) {
+        throw SonataError(fmt::format("expected integer, got float {}", v));
+    }
+    return static_cast<int64_t>(v);
```

Comment:
what happens if the range of int64_t is exceeded?

Reply:
it loops around I guess. I just moved this snippet from `node_sets`. Maybe I can devise a way to check that too

Reply:
done (also for uint). I let this unresolved so you can pass and check the solution

Reply:
> I just moved this snippet from node_sets

That's past me being sloppy, present me remembers to check bounds :)

Thanks for adding the checks, they look good.